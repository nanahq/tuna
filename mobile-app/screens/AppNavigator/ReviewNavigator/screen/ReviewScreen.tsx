import {Text, View} from 'react-native'
import {ReviewsHeader} from "@screens/AppNavigator/ReviewNavigator/screen/components/ReviewsHeader";
import {useSelector} from "react-redux";
import {RootState} from "@store/index";
import {MostReviewedMenus} from "@screens/AppNavigator/ReviewNavigator/screen/components/MostReviewedMenus";
import {tailwind} from "@tailwind";
import {GenericFlashList} from "@components/views/FlashList";
import {useCallback, useRef} from "react";
import {EmptyReviews} from "@screens/AppNavigator/ReviewNavigator/screen/components/EmptyReviews";
import {ReviewCard} from "@screens/AppNavigator/ReviewNavigator/screen/components/ReviewCard";
import {Review} from "@store/reviews.reducer";

const data = [{
    reviewDate: Date.now().toLocaleString(),
    reviewUser: 'Adam',
    reviewDesc: 'Food is phenomenal',
    reviewRating: '4.8',
    reviewListingName: 'Fufu soup'
},
    {
        reviewDate: new Date().getTime().toLocaleString(),
        reviewUser: 'Adam',
        reviewDesc: 'Food is phenomenal',
        reviewRating: '4.8',
        reviewListingName: 'Fufu soup'
    },

    {
        reviewDate: new Date().getTime().toLocaleString(),
        reviewUser: 'Adam',
        reviewDesc: 'Food is phenomenal',
        reviewRating: '4.8',
        reviewListingName: 'Fufu soup'
    },

    {
        reviewDate: new Date().getTime().toLocaleString(),
        reviewUser: 'Adam',
        reviewDesc: 'Food is phenomenal',
        reviewRating: '4.8',
        reviewListingName: 'Fufu soup'
    },


]


export function ReviewScreen (_props: any): JSX.Element {
    const {reviews, hasFetchedReviews, ...rest} = useSelector((state: RootState) => state.reviews)
    const getTopRatedMenus = useCallback(() => {
        return reviews.sort((a, b) => Number(a.reviewRating) - Number(b.reviewRating)).slice(0, 4)
    }, [])
    return  (
            <>
                <ReviewsHeader {...rest} />
                <View style={tailwind('px-3')}>
                    <MostReviewedMenus reviews={data} />
                    <View style={tailwind('mt-5')}>
                        <Text style={tailwind('text-brand-black-500 font-semibold text-lg mb-3')}>Recent Reviews</Text>
                        <RecentViews reviews={data} />
                    </View>
                </View>
            </>
    )
}


function RecentViews (props: {reviews: Review[]}): JSX.Element {
    const ref = useRef<any>(null)
    const RenderItems = useCallback(
        (props: { item: Review; index: number }): JSX.Element => {
            return (
                <ReviewCard
                    {...props.item}

                />
            );
        },
        []
    );
    return (
        <GenericFlashList
            contentContainerStyle={tailwind("pb-2")}
            data={props.reviews}
            ref={ref}
            estimatedItemSize={30}
            keyExtractor={(_item, index) => index.toString()}
            ListEmptyComponent={<EmptyReviews />}
            renderItem={RenderItems}
        />
    );
}


