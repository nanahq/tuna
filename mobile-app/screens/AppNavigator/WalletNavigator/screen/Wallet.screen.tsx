import {Dimensions, Text, View} from 'react-native'
import {tailwind} from '@tailwind'
import {EarningsOVerview} from "@screens/AppNavigator/WalletNavigator/component/EarningsOVerview";
import {FlashList, ListRenderItemInfo} from "@shopify/flash-list";
import {PayoutCard} from "@screens/AppNavigator/WalletNavigator/component/PayoutCard";
import {useSelector} from "react-redux";
import {RootState} from "@store/index";
import { VendorPayoutI } from '@imagyne/eatlater-types';
import { LoaderComponentScreen } from '@components/commons/LoaderComponent';
import { EmptyAnimation } from '@components/lottie/Empty';


export function WalletScreen (): JSX.Element {
    const {payouts, overview, hasFetchedWallet} = useSelector((state: RootState) => state.wallet )
    
    if(!hasFetchedWallet) {
        return (
            <LoaderComponentScreen />
        )
    }
    
    return (
        <View style={tailwind('px-5 flex-1 flex bg-brand-gray-500')}>
            <EarningsOVerview overview={overview} />  

            <Text style={tailwind('text-lg font-semibold text-brand-black-500 mt-4 mb-2')}>Payouts</Text>
            {payouts.length < 1  ? (
                <EmptyAnimation text='No transaction has been made' />
            ): (
                <PayoutHistory payouts={payouts} />
            )}  
        </View>
    )
}

export function PayoutHistory ({payouts}: {payouts: VendorPayoutI[]}): JSX.Element {
    const screen = Dimensions.get('screen')
    const renderItem = ({item, index}: ListRenderItemInfo<VendorPayoutI>) => (
        <PayoutCard payout={item} index={index} />
    )

    return (
        <View style={[tailwind(''), {
            height: screen.height / 2
        }]}>
            <FlashList
                data={payouts}
                renderItem={renderItem}
                keyExtractor={item => item._id}
                estimatedItemSize={30}
                showsVerticalScrollIndicator={false}
            />
        </View>
    )
}
