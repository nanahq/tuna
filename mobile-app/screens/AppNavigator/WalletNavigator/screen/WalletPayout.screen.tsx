import {ScrollView, View, Text} from "react-native";
import {StackScreenProps} from "@react-navigation/stack";
import {WalletParamsList} from "@screens/AppNavigator/WalletNavigator/WalletNavigator";
import {WalletScreenName} from "@screens/AppNavigator/WalletNavigator/WalletScreenName";
import {useEffect} from "react";
import {GoBackButton} from "@screens/AppNavigator/SettingsNavigator/components/Goback";
import {tailwind} from "@tailwind";


type WalletPayoutScreenProps = StackScreenProps<WalletParamsList, WalletScreenName.WALLET_PAYOUT>
export const WalletPayoutScreen: React.FC<WalletPayoutScreenProps> = ({navigation, route}) => {
    useEffect(() => {
        navigation.setOptions({
            headerTitle: `Payout #${route.params.payout.refId}`,
            headerLeft: () => <GoBackButton onPress={() => navigation.goBack()} />
        })
    }, [])
    return (
        <ScrollView style={tailwind('flex-1 bg-white px-5 pt-5')}>
            <View style={tailwind('flex flex-row items-center')}>
                <Text>Orders</Text>
            </View>
        </ScrollView>
    )
}
