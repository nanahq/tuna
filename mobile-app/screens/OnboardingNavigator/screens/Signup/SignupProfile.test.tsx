import {SignupProfileScreen} from './SignupProfile.screen'
import {render} from '@testing-library/react-native'

jest.mock("@react-navigation/native", () => ({
    useNavigation: jest.fn(),
}));

describe('<SignupProfileScreen', () => {
    it('Should match snapshot', () => {
        const navigation: any = {
            navigate: jest.fn(),
        };
        const route: any = {};

        const rendered = render(
            <SignupProfileScreen
                navigation={navigation}
                route={route}
            />
        )

        // @ts-ignore
        expect(rendered.toJSON()).toMatchSnapshot()
    })
})
