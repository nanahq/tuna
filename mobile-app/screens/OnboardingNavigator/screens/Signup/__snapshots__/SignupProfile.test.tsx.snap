// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupProfileScreen Should match snapshot 1`] = `
<RCTScrollView
  style={
    [
      {
        "display": "flex",
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "100%",
      },
    ]
  }
  testId="SignupProfileScreen.View"
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 40,
          "width": "100%",
        }
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
              "flexDirection": "column",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "rgba(30, 35, 39, 1)",
                "fontFamily": "MediumFont",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginBottom": 10,
              },
              undefined,
            ]
          }
          testID="SignupProfileScreen.FirstName.Label"
        >
          First Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            [
              {
                "backgroundColor": "rgba(219, 225, 229, 1)",
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "color": "rgba(30, 35, 39, 1)",
                "fontFamily": "MediumFont",
                "fontSize": 18,
                "fontWeight": "500",
                "lineHeight": 28,
                "paddingBottom": 14,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 14,
              },
              {
                "width": 160,
              },
            ]
          }
          testID="SignupProfileScreen.FirstName.Input"
          value=""
        />
      </View>
      <View
        style={
          [
            {
              "display": "flex",
              "flexDirection": "column",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "rgba(30, 35, 39, 1)",
                "fontFamily": "MediumFont",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 20,
                "marginBottom": 10,
              },
              undefined,
            ]
          }
          testID="SignupProfileScreen.LastName.Label"
        >
          Last Name
        </Text>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            [
              {
                "backgroundColor": "rgba(219, 225, 229, 1)",
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "color": "rgba(30, 35, 39, 1)",
                "fontFamily": "MediumFont",
                "fontSize": 18,
                "fontWeight": "500",
                "lineHeight": 28,
                "paddingBottom": 14,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 14,
              },
              {
                "width": 160,
              },
            ]
          }
          testID="SignupProfileScreen.LastName.Input"
          value=""
        />
      </View>
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "column",
          },
          {
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.PhoneNumber.Label"
      >
        Phone Number
      </Text>
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(219, 225, 229, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 18,
              "fontWeight": "500",
              "lineHeight": 28,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.PhoneNumber.Input"
        value=""
      />
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "column",
          },
          {
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.Email.Label"
      >
        Email
      </Text>
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(219, 225, 229, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 18,
              "fontWeight": "500",
              "lineHeight": 28,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.Email.Input"
        value=""
      />
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "column",
          },
          {
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.Password.Label"
      >
        Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(219, 225, 229, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 18,
              "fontWeight": "500",
              "lineHeight": 28,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.Password.Input"
        value=""
      />
    </View>
    <View
      style={
        [
          {
            "display": "flex",
            "flexDirection": "column",
          },
          {
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 20,
              "marginBottom": 10,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.ConfirmPassword.Label"
      >
        Confirm Password
      </Text>
      <TextInput
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        style={
          [
            {
              "backgroundColor": "rgba(219, 225, 229, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "color": "rgba(30, 35, 39, 1)",
              "fontFamily": "MediumFont",
              "fontSize": 18,
              "fontWeight": "500",
              "lineHeight": 28,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
            },
            undefined,
          ]
        }
        testID="SignupProfileScreen.ConfirmPassword.Input"
        value=""
      />
    </View>
    <View
      accessibilityState={
        {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "rgba(115, 115, 115, 1)",
          "borderBottomLeftRadius": 12,
          "borderBottomRightRadius": 12,
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "SemiBoldFont",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
              "paddingBottom": 14,
              "paddingTop": 14,
              "textAlign": "center",
            },
            {
              "color": "rgba(255, 255, 255, 1)",
            },
          ]
        }
      >
        Continue
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginTop": 32,
          "opacity": 1,
        }
      }
      testID="LoginButtonWithText.Button"
    >
      <Text
        style={
          {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "RegularFont",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
            "textAlign": "center",
          }
        }
        testID="LoginButtonWithText.Text"
      >
        Do you have an account? Login
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
